DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/searchSortFilterAndPaginate.ts

```diff
@@ -107,9 +107,11 @@ export class SearchSortFilterAndPaginate {
     const sortedElements = sortSearchableElements(filteredElements);
 
     const sortedUniqueElements = _.uniqBy(sortedElements, 'parentIndex');
-    const resultCount = sortedUniqueElements.length;
-    SearchSortFilterAndPaginate
-      .updateResultCount(resultCount);
+
+    if (updateResultCount) {
+      const resultCount = sortedUniqueElements.length;
+      SearchSortFilterAndPaginate.updateResultCount(resultCount);
+    }

```

Comment:
This has been performed so that the result count is not updated when updateResultCount is false. It is only set to false in the constructor when the page is initially loading. This prevents the result count being announced twice on any page load when JS is enabled.

The reason for this is twofold:

1. When we hit this in the constructor, the page should already have its result count set from the basic GET request (the same that occurs without JS). This shouldn't need to change when JS loads its data, so this is unnecessary to run.
2. The result count was being read out twice on page load because the result count has an alert role. It was read out when the page was initially displayed, and then read out again when the JS loading updated it to the same value. Normally, alerts do not get announced if the text inside is updated to the exact same value. However, we add a non-breaking space to any text value that is identical when updating, see https://github.com/TechnologyEnhancedLearning/DLSV2/pull/648 for further details.